### PR TITLE
Overboard fixes

### LINKED ISSUE #113: Enable overboard

### PR DESCRIPTION
**Pull request by nonmakina**
Removes the random capability from the overboard and fixes the thread preview generator.
It used to show posts that were in the middle of the thread, now it shows the last posts.

### CODE REVIEW

File: templates/themes/semirand/theme.php [hunk 1]
```diff
@@ -131,13 +133,13 @@ private function buildOne($post, $mod = false) {
 
 			openBoard($post['board']);
 			$thread  = new Thread($post, $mod ? '?/' : $config['root'], $mod);
-			$replies = $this->fetchReplies($post['board'], $post['id']);
 			// Number of replies to a thread that are displayed beneath it
 			$preview_count = $post['sticky'] ? $config['threads_preview_sticky'] :
 				$config['threads_preview'];
+			$replies = $this->fetchReplies($post['board'], $post['id'], $preview_count);
```
Comment (towards-a-new-leftypol) [inline templates/themes/semirand/theme.php:139]: :+1:

**Comment by nonmakina** [comment]
Overboard has link to catalog, but the link is broken.

**Comment by nonmakina** [comment]
Overboard link thing has a problem.

**Comment by nonmakina** [comment]
Possibly related to the relative filepath issue. 
https://github.com/towards-a-new-leftypol/leftypol_lainchan/issues/124

File: inc/display.php [hunk 2]
```diff
@@ -66,8 +66,27 @@ function createBoardlist($mod=false) {
     foreach ($xboards as $val) {
         $boards[$val['uri']] = $val['title'];
     }
+    $body = '';
+    if (isset($config['prepended_foreign_boards'])){
```
Comment (towards-a-new-leftypol) [inline inc/display.php]: You shouldn't copy paste this block, please refactor this out to it's own function.

**Review by towards-a-new-leftypol — CHANGES_REQUESTED** [review]
Overall really fucking great! We just need to fix the DRY principle violation and the link at the top in instance-config.

File: inc/instance-config.php [hunk 3]
```diff
@@ -23,6 +23,11 @@
    ) ,
   array('meta')
 );
+
+$config['prepended_foreign_boards'] = array(
+    'overboard' => '/overboard',
```
Comment (towards-a-new-leftypol) [inline inc/instance-config.php]: missing a trailing slash (link at the top on dev does not work without this)

**Review by towards-a-new-leftypol — DISMISSED** [review]

**Review by towards-a-new-leftypol — APPROVED** [review]

**Pull request merged**